datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -937,7 +944,8 @@ impl GroupedHashAggregateStream {
         // over the target memory size after emission, we can emit again rather than returning Err.
         let _ = self.update_memory_reservation();
         let batch = RecordBatch::try_new(schema, output)?;
-        Ok(batch)
+        debug_assert!(batch.num_rows() > 0);
```

Comment:
Maybe it would be good to document in comments somewhere the expectation / behavior that no empty record batches are produced